src/seat.c

```diff
@@ -506,13 +506,51 @@ seat_finish(struct server *server)
 }
 
 static void
-configure_keyboard(struct wlr_input_device *device)
+reset_window_keyboard_layout_groups(struct server *server)
 {
+	if (!rc.kb_layout_per_window) {
+		return;
+	}
+
+	/*
+	 * We don't want to reset all window keyboard-layout groups on every
+	 * reconfigure unless XKB_DEFAULT_LAYOUT has actully changed.
+	 */
+	static char last_layout[64];
```

Comment:
I am not sure if 64 is enough. You can put of \*lot\* of options there.

Reply:
Have increased to 1024.